external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaContinuousWriter.scala

```diff
@@ -112,8 +112,8 @@ class KafkaContinuousDataWriter(
     checkForErrors()
     if (producer != null) {
       producer.flush()
+      producer.inUseCount.decrementAndGet()
       checkForErrors()
-      CachedKafkaProducer.close(new java.util.HashMap[String, Object](producerParams.asJava))
```

Comment:
Since a producer is shared across threads, we maintain inuse counts and close them separately.